tools/validate_configs/validate_configs.sh

```diff
@@ -21,16 +21,26 @@ run_test() {
 	exampleFile=$(basename "$example")
 	DEPLOYMENT=$(echo "${exampleFile%.yaml}-$(basename "${tmpdir##*.}")" | sed -e 's/\(.*\)/\L\1/')
 	PROJECT="invalid-project"
+	LOCAL_SOURCE_PATTERN='source: ./'
 
 	echo "testing ${example} in ${tmpdir}"
 	cp "${example}" "${tmpdir}/"
-	cd "${cwd}"
-	./ghpc create -l IGNORE --vars "project_id=${PROJECT},deployment_name=${DEPLOYMENT}" "${tmpdir}"/"${exampleFile}" >/dev/null ||
+
+	# Only run from the repo directory if there are local modules, otherwise
+	# run the test from the test directory using the installed ghpc binary.
+	if grep -q "${LOCAL_SOURCE_PATTERN}" "${cwd}/${example}"; then
+		cd "${cwd}"
+	else
+		cd "${tmpdir}"
+	fi
+	ghpc create -l IGNORE --vars "project_id=${PROJECT},deployment_name=${DEPLOYMENT}" "${tmpdir}"/"${exampleFile}" >/dev/null ||
```

Comment:
Consider this can also run on the user development machine (their laptops or desktops). I frequently call make tests from my workstation and this is a use case we want to continue supporting. 

This line is testing the system's ghpc (from the path), and not the local binary that you are building. Tests should focus at what you are specifically building and testing (Tests should not have system wide effects).

You should, instead, call ${cwd}/ghpc which is the one you just built.

Reply:
I agree with this concern and didn't properly consider it since make install copies into my PATH even as a non-superuser. +1

Reply:
Done